### PR TITLE
Fix some broken URLs in KataConfig description

### PR DESCRIPTION
**Pull request by gkurz**
Trivial update in code comments to use the appropriate doc link.

### CODE REVIEW

**Comment by apinnick** [comment]
I created https://issues.redhat.com/browse/HCIDOCS-463 to set up permanent anchors in the docs for NFD and peer pods. 

@gkurz Can you create a variable for the OSC version number in the URL?

File: api/v1/kataconfig_types.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ type KataConfigSpec struct {
 
 	// CheckNodeEligibility is used to detect the node(s) eligibility to run Kata containers.
 	// This is currently done through the use of the Node Feature Discovery Operator (NFD).
-	// For more information on how the check works, please refer to the sandboxed containers documentation - https://docs.openshift.com/container-platform/latest/sandboxed_containers/deploying-sandboxed-container-workloads.html
+	// For more information on how the check works, please refer to the sandboxed containers documentation - https://docs.redhat.com/en/documentation/openshift_sandboxed_containers/1.6/html/user_guide/deploying-bare-metal
```
Comment (apinnick) [inline api/v1/kataconfig_types.go]: https://docs.redhat.com/en/documentation/openshift_sandboxed_containers/1.6/html-single/user_guide/index#about-node-eligibility-checks_about-osc

File: api/v1/kataconfig_types.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ type KataConfigSpec struct {
 	LogLevel string `json:"logLevel,omitempty"`
 
 	// EnablePeerPods is used to transparently create pods on a remote system.
-	// For more information on how this works, please refer to the sandboxed containers documentation - https://docs.openshift.com/container-platform/latest/sandboxed_containers/deploying-sandboxed-container-workloads.html
+	// For more information on how this works, please refer to the sandboxed containers documentation - https://docs.redhat.com/en/documentation/openshift_sandboxed_containers/1.6/html/user_guide/deploying-public-cloud
```
Comment (apinnick) [inline api/v1/kataconfig_types.go]: https://docs.redhat.com/en/documentation/openshift_sandboxed_containers/1.6/html/user_guide/deploying-public-cloud#deploying-public-cloud

**Review by apinnick — CHANGES_REQUESTED** [review]

**Comment by gkurz** [comment]
Thanks @apinnick ! I've updated the links according to your suggestion. Please approve.

It is a bit unfortunate that the publication system doesn't support a "latest" version. For the time being, it is acceptable to point to a numerical version as long as the steps documented there remain valid but it would be great to find a way to have permanent links.

**Review by bpradipt — APPROVED** [review]
/lgtm

File: api/v1/kataconfig_types.go [hunk 3]
```diff
@@ -31,7 +31,7 @@ type KataConfigSpec struct {
 
 	// CheckNodeEligibility is used to detect the node(s) eligibility to run Kata containers.
 	// This is currently done through the use of the Node Feature Discovery Operator (NFD).
-	// For more information on how the check works, please refer to the sandboxed containers documentation - https://docs.openshift.com/container-platform/latest/sandboxed_containers/deploying-sandboxed-container-workloads.html
+	// For more information on how the check works, please refer to the sandboxed containers documentation - https://docs.redhat.com/en/documentation/openshift_sandboxed_containers/1.6/html-single/user_guide/index#about-node-eligibility-checks_about-osc
```
Comment (bpradipt) [inline api/v1/kataconfig_types.go:34]: assuming 1.6 is intentional here

Comment (apinnick) [reply api/v1/kataconfig_types.go:34]: Unfortunately, that is a publishing platform limitation. Currently, version numbers are hard-coded in URLs. Maybe the new tooling platform will allow more flexibility.

**Pull request merged**